test/hotreload/myrole_test.lua

```diff
@@ -0,0 +1,398 @@
+local fio = require('fio')
+local yaml = require('yaml')
+local fiber = require('fiber')
+local httpc = require('http.client')
+local socket = require('socket')
+local t = require('luatest')
+local g = t.group()
+
+local helpers = require('test.helper')
+
+local function reload_myrole(fn)
```

Comment:
I would be great to have a comment (for future) that this approach doesn't work with local variables. And we should use only globals inside. With link to https://www.lua.org/manual/5.1/manual.html#pdf-string.dump (*function must be a Lua function without upvalues*)